gms/gossiper.hh

```diff
@@ -188,6 +188,10 @@ public:
     // value this node would get if this node were restarted that we are
     // willing to accept about a peer.
     static constexpr int64_t MAX_GENERATION_DIFFERENCE = 86400 * 365;
+    // Maximum queue size on _apply_state_locally_semaphore.
+    // After this queue size is reached, we drop non-critical updates.
+    static constexpr size_t apply_max_queue_size{1000};
```

Comment:
I am not sure about the number. It looks reasonable though.

Reply:
I just pulled it out of thin air. Should it be lower perhaps? In the concrete case where I saw the huge message queue causing OOM, I saw 65K messages queued up, I chose one order of magnitude smaller.
I think we can do 100 instead. A large queue is not useful, better shed messages sooner and get back to processing fresh messages faster.